numpy/distutils/fcompiler/gnu.py

```diff
@@ -200,9 +202,17 @@ def get_flags_opt(self):
             # With this compiler version building Fortran BLAS/LAPACK
             # with -O3 caused failures in lib.lapack heevr,syevr tests.
             opt = ['-O2']
+        elif v and v>='4.6.0':
+            if is_win32():
+                # use -mincoming-stack-boundary=2
+                # due to the change to 16 byte stack alignment since GCC 4.6
+                # but 32 bit Windows ABI defines 4 bytes stack alignment
+                opt = ['-O2 -march=core2 -mtune=generic -mfpmath=sse -msse2 -mincoming-stack-boundary=2']
+            else:
+                opt = ['-O2 -march=x86-64 -DMS_WIN64 -mtune=generic -msse2']
```

Comment:
This seems to add the `-DMS_WIN64` and other flags to all platforms, including all non-windows ones